mopidy/mopidy
# Cherry-picked fixes for a 3.4.2 bugfix release

- Upgrade actions/checkout from v2 to v3
- Upgrade actions/setup-python from v2 to v4
- Upgrade actions/cache from v2 to v3
- Upgrade host OS from Ubuntu 20.04 to 22.04
- Format with Black 23.1.0
- Set warnings.warn() stacklevel to improve stacktrace usability
- Improve Exception test
- Add Python 3.11 to test matrix
- Upgrade codecov/codecov-action from v1 to v3
- Upgrade pypa/gh-action-pypi-publish from 1.5.0 to 1.8.1
- Only run workflows once when pushing to a PR branch
- Fix extlinks crash in docs build
- Upgrade CodeQL actions
- Use setup-python's builtin cache support
- Add Python 3.11 to PyPI classifiers
- Update GitHub Actions badge
- Build docs using Python 3.11
- Update copyright years
- Reencode test files to fix duration mismatch
- Update README with developer info. (#2090)
- Fix README.rst typo
- Update README.rst (#2091)
- Formatting and punctuation
- file: Correct file/follow_symlinks error. (#2094)
- tests: Allow tests to run on MacOS. (#2092)
- docs: remove references to python 3.7. (#2098)
- Added tests to ensure 100% of Mopidy-file. (#2096)
- tox -e flake8 failure solved. (#2101)
- Set codecov threshold to 0.3% before failing.
- add "tox -e ci", to test before pushing. (#2102)
- add "tox -e ci" to developer doc. (#2105)
- Remove CodeQL analysis
- docs: Install graphviz for rendering state diagrams
- Update README.rst (#2129)
- Apply grammer fixes from #2129 to docs too
- Update changelog
